src/kwio.sh

```diff
@@ -119,3 +119,36 @@ function ask_yN()
     printf '%s\n' '0'
   fi
 }
+
+# Asks for user input
+#
+# @message A string with the message to be displayed to the user.
+# @default_option A string with the default answer of the user (if the user simply
+#                 inputs nothing, this will be considered the answer)
+# @show_default A boolean value that defines if the default value will be printed
+#               to the user. If this value is empty, this will be considered true.
+#
+# Returns: The user answer, garanted it is not empty.
```

Comment:
There is a typo here, I'm not sure what you meant, but I think it was either `granted` or `guaranteed`